montepy/mcnp_problem.py

```diff
@@ -390,25 +391,58 @@ def add_cell_children_to_problem(self):
         self._transforms = Transforms(transforms)
         self._data_inputs = sorted(set(self._data_inputs + materials + transforms))
 
-    def write_to_file(self, new_problem, overwrite=False):
+    def write_problem(self, destination, overwrite=False):
+        """
+        Write the problem to a file or writeable object.
+
+        :param destination: File path or writable object
+        :type destination: {io.TextIOBase, str, os.PathLike}
+        :param overwrite: Whether to overwrite 'destination' if it is an existing file
+        :type overwrite: bool
+        """
+        if hasattr(destination, "write") and callable(getattr(destination, "write")):
+            new_file = MCNP_InputFile.from_open_stream(destination)
+            self._write_to_stream(new_file)
+        elif isinstance(destination, (str, os.PathLike)):
+            new_file = MCNP_InputFile(destination, overwrite=overwrite)
+            with new_file.open("w") as fh:
+                self._write_to_stream(fh)
+        else:
+            raise TypeError(
+                f"destination f{destination} is not a file path or writable object"
+            )
+
+    def write_to_file(self, file_path, overwrite=False):
         """
         Writes the problem to a file.
 
+        .. versionchanged:: 0.4.0
+            Deprecated in favor of self.write_problem()
+
         .. versionchanged:: 0.3.0
             The overwrite parameter was added.
 
-        :param new_problem: the file name to write this problem to
-        :type new_problem: str
+        :param file_path: the file path to write this problem to
+        :type file_path: {str, os.PathLike}
         :param overwrite: Whether to overwrite the file at 'new_problem' if it exists
         :type overwrite: bool
         :raises IllegalState: if an object in the problem has not been fully initialized.
         :raises FileExistsError: if a file already exists with the same path.
         :raises IsADirectoryError: if the path given is actually a directory.
         """
-        new_file = MCNP_InputFile(new_problem, overwrite=overwrite)
-        with new_file.open("w") as fh, warnings.catch_warnings(
-            record=True
-        ) as warning_catch:
+        cn = self.__class__.__name__
+        msg = f"{cn}.write_to_file() is deprecated. Use {cn}.write_problem()"
+        warnings.warn(msg, DeprecationWarning)
```

Comment:
I'm not sure if we need to raise deprecation warnings yet.

Reply:
`PendingDeprecationWarning`, or no warning at all?

Reply:
pending is an option? I'm trying to think: what would @bascandr say?

I honestly don't see any reason to truly deprecate it and remove it. I think no warning. I'm fine with users not making changes at this point.